### ISSUE TITLE
webContents will-redirect not working

### ISSUE THREAD

**Issue by adammeyer** [electron/electron#15684]
* Output of `node_modules/.bin/electron --version`: v2.0.13.
* Operating System (Platform and Version): MacOS
* Output of `node_modules/.bin/electron --version` on last known working Electron version (if applicable): Unknown

**Expected Behavior**
Event to trigger.

**Actual behavior**
Event is never trigger.  `did-navigate` works fine, `will-redirect` does not.

**To Reproduce**
This redirects to https://www.google.com, but no event.
```
let window = new electron.BrowserWindow({});
window.webContents.on('will-redirect', (event, url) => {
  console.log('Redirect', url);
});
window.loadURL('http://www.google.com');
```

**Comment by adammeyer** [comment]
I noticed `will-redirect` isn't in this list.  https://github.com/electron/electron/blob/2-1-x/lib/browser/guest-view-manager.js#L9

**Comment by MarshallOfSound** [comment]
@adam-lynch `will-redirect` was only implemented for `4-0-x` 👍 If you use Electron `4.0.0-beta.X` that event will be there 👍

**Comment by oula-oula** [comment]
sorry i didn't understand
